lib/ansible/modules/windows/win_file.ps1

```diff
@@ -20,51 +20,601 @@
+        }
+
+        private static void CreateLinkHard(string linkPath, String target)
+        {
+            bool success = CreateHardLink(linkPath, target, IntPtr.Zero);
+            if (!success)
+                throw new Exception(String.Format("Failed to create Hard Link {0} to Target {1}: {2}", linkPath, target, new Win32Exception(Marshal.GetLastWin32Error()).Message));
+        }
+
+        private static void CreateJunctionPoint(string linkPath, string target)
+        {
+            Directory.CreateDirectory(linkPath);
+            SafeFileHandle handle = CreateFile(linkPath, FileAccess.Write, FileShare.Read | FileShare.Write | FileShare.None, IntPtr.Zero, FileMode.Open, FILE_FLAG_BACKUP_SEMANTICS | FILE_FLAG_OPEN_REPARSE_POINT, IntPtr.Zero);
+            if (handle.IsInvalid)
+                throw new Exception(String.Format("Unable to open Reparse Point at {0}: {1}", linkPath, new Win32Exception(Marshal.GetLastWin32Error()).Message));
+
+
+            string substituteName = MAXIMUM_PATH_FIX_PREFIX + Path.GetFullPath(target);
+            string printName = target;
+
+            short substituteNameOffset = 0;
+            short substituteNameLength = (short)(substituteName.Length * SIZE_OF_WCHAR);
+
+            short printNameOffset = (short)(substituteNameLength + 2);
+            short printNameLength = (short)(printName.Length * SIZE_OF_WCHAR);
+
+            short reparseDataLength = (short)(substituteNameLength + printNameLength + 12);
+
+            byte[] unicodeBytes = Encoding.Unicode.GetBytes(substituteName + "\0" + printName);
+            char[] pathBuffer = Encoding.Unicode.GetChars(unicodeBytes);
+
+            REPARSE_DATA_BUFFER reparseDataBuffer = new REPARSE_DATA_BUFFER();
+            reparseDataBuffer.ReparseTag = IO_REPARSE_TAG_MOUNT_POINT;
+            reparseDataBuffer.ReparseDataLength = reparseDataLength;
+            reparseDataBuffer.SubstituteNameOffset = substituteNameOffset;
+            reparseDataBuffer.SubstituteNameLength = substituteNameLength;
+            reparseDataBuffer.PrintNameOffset = printNameOffset;
+            reparseDataBuffer.PrintNameLength = printNameLength;
+            reparseDataBuffer.PathBuffer = new char[MAXIMUM_REPARSE_DATA_BUFFER_SIZE];
+
+            Array.Copy(pathBuffer, reparseDataBuffer.PathBuffer, pathBuffer.Length);
+            int inBufferSize = Marshal.SizeOf(reparseDataBuffer);
+            IntPtr inBuffer = Marshal.AllocHGlobal(inBufferSize);
+
+            try
+            {
+                Marshal.StructureToPtr(reparseDataBuffer, inBuffer, false);
+                uint bytesReturned;
+                uint bufferSize = (uint)(reparseDataBuffer.ReparseDataLength + REPARSE_DATA_BUFFER_HEADER_SIZE);
+                REPARSE_DATA_BUFFER outBuffer = new REPARSE_DATA_BUFFER();
+                bool success = DeviceIoControl(handle, FSCTL_SET_REPARSE_POINT, inBuffer, bufferSize, out outBuffer, 0, out bytesReturned, IntPtr.Zero);
```

Comment:
Ah, old P/Invoke tricks: because it's blittable, you should be able to change the definition to `class REPARSE_DATA_BUFFER` instead of `struct`- then you can pass null. :)